types/select2/v3/index.d.ts

```diff
@@ -168,6 +168,14 @@ interface Select2Plugin {
     (method: 'search'): JQuery;
 
     (options: Select2Options): JQuery;
+
+    /**
+     * Select2 exposes its default options via the $.fn.select2.defaults
+     * object. Properties changed in this object (same properties configurable
+     * through the constructor) will take effect for every instance created
+     * after the change.
+     */
+    defaults: Partial<Select2Options>;
```

Comment:
Unfortunately this does not work beacuase of the nested options: https://select2.org/configuration/defaults#nested-options.

Reply:
My change applies to the `v3/` dir, i.e. the v3.x branch.
In particular, this is documented here (unfortunately no document anchors, however it is the last paragraph in the page):

http://select2.github.io/select2/

> Select2 exposes its default options via the `$.fn.select2.defaults` object. Properties changed in this object (same properties configurable through the constructor) will take effect for every instance created after the change.

Note the wording _same properties configurable through the constructor_ which is why I assumed it'd be safe to use `Select2Options`.